src/storage/mod.rs

```diff
@@ -1295,12 +1295,22 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
     ) -> Result<()> {
         check_key_size!(Some(&key).into_iter(), self.max_key_size, callback);
 
+        let convert_to_write = true;
+        let m: Modify = if convert_to_write {
+            // let empty_val: Vec<u8> = vec![];
+            let mut m_tmp = Modify::Put(
+                Self::rawkv_cf(&cf)?,
+                Key::from_encoded(key), vec![]);
+            m_tmp.with_extended_fields(convert_to_expire_ts(30), None, true);
+            m_tmp
+        } else {
+            Modify::Delete(
+                Self::rawkv_cf(&cf)?,
+                Key::from_encoded(key))
+        };
```

Comment:
We need a callback to ingest a timestamp in `pre_propose` phrase. And the `TTL` is designed to be 24 hours for the deleted entry to be garbage collected.
You can refer to RFC and codes in `raw_put`.